sw/device/lib/dif/dif_spi_device.c

```diff
@@ -1550,7 +1550,8 @@ dif_result_t dif_spi_device_tpm_write_data(dif_spi_device_handle_t *spi,
   if (result != kDifOk) {
     return result;
   }
-  if (DIF_SPI_DEVICE_TPM_FIFO_DEPTH - status.read_fifo_occupancy < length) {
+  if ((DIF_SPI_DEVICE_TPM_FIFO_DEPTH - status.read_fifo_occupancy) * sizeof(uint32_t) <
```

Comment:
@a-will can you have a look at this part? 
There was an issue with width mix-up before. The `TPM_FIFO_DEPTH` and `read_fifo_occupancy` were all using 4B as its width, while the length was using number of bytes.

Reply:
DIF_SPI_DEVICE_TPM_FIFO_DEPTH was actually in bytes, but it needs to be updated after [cb165ba7f497553e6d461a7abb894987c006371d](https://github.com/lowRISC/opentitan/pull/14544). There was a to-do item in that PR, looks like, to change the API to take in only word-aligned buffers as well.

Reply:
wait, but does that mean it used to be 4 bytes? 
anyways, i tweaked it a bit. let me know if this works for now.